### DISCUSSION TITLE
Verifying the signature on the certificate

### DISCUSSION THREAD

**Discussion by ssokram** [Intermesh/groupoffice#794] [General]
When I try to verify the certificate signature of a signed message I get the following error message:
Exception in /usr/share/groupoffice/go/base/Model.php at line 109: Can't set not existing property 'ocspMsg' in 'GO\Smime\Model\Certificate'.

The certificate used is generated from AD CA and based on Exchange Template. If I generate a certificate based on the User Template everything works fine.

Any help is appreciated.

**Comment by mschering** [comment]
Which version do you use? I can't replicate this in 6.6.

**Reply by ssokram** [reply]
The version is 6.6.35.  As I mentioned the problem only occurs when I use a certificate based on Exchange Template.  I also tested my Exchange certificate with the SOGO mail client and everything works well, so I assume that the certificate is ok.

**Comment by mschering** [comment]
Can you enable debug in config.php? Set $config['debug'] = true; then post the stack trace of the new error message here.

**Reply by ssokram** [reply]
Error from debug.log:

[log][GO\Email\Model\Account:392] imap::connect(-----------------, 993, --------------, ***, 1, )
[log][GO\Base\Mail\Imap:142] Use capability from login
[log][GO\Base\Mail\Imap:283] Use capability from login
[log][GO\Email\Model\Account:373] Selecting IMAP mailbox INBOX
[log][GO\Base\Mail\Imap:2452] Part size: 923
[log][GO\Base\Mail\Imap:2584] get_message_part_decoded(5088, TEXT, , )
[log][GO\Base\Mail\Imap:2309] Part size: 8047
[log][GO\Base\Router:186] 
        EXCEPTION: Exception: Can't set not existing property 'ocspMsg' in 'GO\Smime\Model\Certificate' in /usr/share/groupoffice/go/base/Model.php:109
        Stack trace:
        #0 /usr/share/groupoffice/go/base/db/ActiveRecord.php(4543): GO\Base\Model->__set()
        #1 /usr/share/groupoffice/go/base/db/ActiveRecord.php(4387): GO\Base\Db\ActiveRecord->setAttribute()
        #2 /usr/share/groupoffice/modules/smime/model/PublicCertificate.php(136): GO\Base\Db\ActiveRecord->__set()
        #3 /usr/share/groupoffice/modules/smime/controller/PublicCertificateController.php(64): GO\Smime\Model\PublicCertificate::fromEmail()
        #4 /usr/share/groupoffice/go/base/controller/AbstractController.php(500): GO\Smime\Controller\PublicCertificateController->actionVerify()
        #5 /usr/share/groupoffice/go/base/controller/AbstractController.php(398): GO\Base\Controller\AbstractController->callActionMethod()
        #6 /usr/share/groupoffice/go/base/Router.php(186): GO\Base\Controller\AbstractController->run()
        #7 /usr/share/groupoffice/index.php(115): GO\Base\Router->runController()
        #8 {main}
[error][go\core\ErrorHandler:68] Exception in /usr/share/groupoffice/go/base/Model.php at line 109: Can't set not existing property 'ocspMsg' in 'GO\Smime\Model\Certificate'
[error][go\core\ErrorHandler:77] #0 /usr/share/groupoffice/go/base/db/ActiveRecord.php(4543): GO\Base\Model->__set()
[error][go\core\ErrorHandler:77] #1 /usr/share/groupoffice/go/base/db/ActiveRecord.php(4387): GO\Base\Db\ActiveRecord->setAttribute()
[error][go\core\ErrorHandler:77] #2 /usr/share/groupoffice/modules/smime/model/PublicCertificate.php(136): GO\Base\Db\ActiveRecord->__set()
[error][go\core\ErrorHandler:77] #3 /usr/share/groupoffice/modules/smime/controller/PublicCertificateController.php(64): GO\Smime\Model\PublicCertificate::fromEmail()
[error][go\core\ErrorHandler:77] #4 /usr/share/groupoffice/go/base/controller/AbstractController.php(500): GO\Smime\Controller\PublicCertificateController->actionVerify()
[error][go\core\ErrorHandler:77] #5 /usr/share/groupoffice/go/base/controller/AbstractController.php(398): GO\Base\Controller\AbstractController->callActionMethod()
[error][go\core\ErrorHandler:77] #6 /usr/share/groupoffice/go/base/Router.php(186): GO\Base\Controller\AbstractController->run()
[error][go\core\ErrorHandler:77] #7 /usr/share/groupoffice/index.php(115): GO\Base\Router->runController()
[error][go\core\ErrorHandler:77] #8 {main}
[log][GO\Base\Db\Columns:69] Using \GO\Base\Cache\Apcu cache
[log][GO:13] [2022-01-28 11:18] INIT POST /groupoffice/index.php?r=summary/announcement/store&security_token=xCbutajy96zDHL54kwd8
[log][GO\Base\Router:125] Controller route r=summary/announcement/store
